tools/extract-verification-info.sh

```diff
@@ -3,14 +3,42 @@
 img=$1
 
 if [ "$img" == "" ]; then
-  echo "Usage ./extract_verification_info.sh <verification.img>"
+  echo "Usage ./extract-verification-info.sh <USB_device_name|verification_img>"
+  echo "E.g., ./extract-verification-info.sh /dev/disk4"
+  echo "Or, ./extract-verification-info.sh dist/amd64/current/verification.raw"
   exit
 fi
 
-size=$(fdisk -l "$img" | grep raw5 | awk '{print $2}')
-mkdir /tmp/verification_mnt
-sudo mount -o offset=$((size*512)) "$img" /tmp/verification_mnt
-cp /tmp/verification_mnt/*/summary.log ./
-cat /tmp/verification_mnt/*/summary.log
-sudo umount /tmp/verification_mnt
-rmdir /tmp/verification_mnt
+checkOScmd="echo ${OSTYPE} | grep -q darwin" # Running on MacOS
+checkVerificationImg="eval file ${img} | grep -q DOS/MBR" # this is a file not a block device
+mountDir="/tmp/verification_mnt"
+if eval "${checkOScmd}"; then # MacOS
+  devicename="${img}"
+  if eval "${checkVerificationImg}"; then # file
+    tmp=$(/usr/bin/hdiutil attach -imagekey diskimage-class=CRawDiskImage -nomount "${img}")
+    devicename=$(echo "${tmp}" | grep "GUID_partition_scheme" | awk '{print $1}')
+  fi
+  /usr/sbin/diskutil mount "${devicename}"s5
+  mountDir="/Volumes/INVENTORY"
+else # Linux
+  mkdir "${mountDir}"
+  if eval "${checkVerificationImg}"; then # file
+    size=$(fdisk -l "$img" | grep raw5 | awk '{print $2}')
+    sudo mount -o offset=$((size*512)) "$img" "${mountDir}"
```

Comment:
You might face some systems were sudo is not available (or not configured for the running user) but the user can still run the script as root, so I think it's better to instruct the user to run the whole script as sudo (or as root) instead of integrate the sudo commands inside the script....

Reply:
@jsfakian I see you've added the "sudo" to the "usage" output and removed it from the commands, but still the check that $USER is root is missing. Could you please add this check at the beginning of the script and if user is not root, then show some error or usage text.

Reply:
Done, thx.